apps/mobile/src/queries/activity/account-activity.query.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  useTotalUnifiedAccountIdentifiers,
+  useUnifiedAccountIdentifier,
+} from '@/hooks/use-account-unified-identifier';
+import { toFetchState } from '@/shared/fetch-state';
+import { useSettings } from '@/store/settings/settings';
+import { QueryFunctionContext, useQuery } from '@tanstack/react-query';
+
+import { UnifiedAccountIdentifier, getActivityService } from '@leather.io/services';
+
+export function useTotalActivity() {
+  const accountIdentifiers = useTotalUnifiedAccountIdentifiers();
+  return toFetchState(useAggregateActivityQuery(accountIdentifiers));
+}
+
+export function useAccountActivity(fingerprint: string, accountIndex: number) {
+  const accountIdentifier = useUnifiedAccountIdentifier(fingerprint, accountIndex);
+  return toFetchState(useAccountActivityQuery(accountIdentifier));
+}
+
+export function useAccountActivityQuery(account: UnifiedAccountIdentifier) {
+  const { fiatCurrencyPreference } = useSettings();
```

Comment:
Seeing this here now too and it's actually great to have gotten this done now as it will be pre-built for our next tasks from the start